### PR TITLE
[Performance] Use new Azure custom images

### PR DESCRIPTION
**Pull request by yika-luo**
Use the new custom image for Azure's VM creation. 

Example performance change (with fixed region):
|  VM Type 💻   | Old Provision 🕐                 | New Provision 🕐     | % Speedup ✅ |
| --------------- | ---------------------------- | ----------- | ----------- |
| Azure GPU      |   4min    | 2min 40s   | 35% (1.5x) |
| Azure CPU       |  4min  | 2min |  50% (2x) |

Example benchmark commands:
CPU: `time sky launch --cloud=azure --region=centralus -y`
GPU: `time sky launch --cloud=azure --region=centralus -y --gpus=A100-80GB:1`

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Relevant individual smoke tests: 
`pytest tests/test_smoke.py::test_minimal --azure`
`pytest tests/test_smoke.py::test_huggingface --azure`
`pytest tests/test_smoke.py::test_job_queue_with_docker --azure`
`pytest tests/test_smoke.py::test_cancel_azure`

### CODE REVIEW

File: sky/clouds/azure.py [hunk 1]
```diff
@@ -39,9 +39,9 @@
 _DEFAULT_AZURE_UBUNTU_2004_IMAGE_GB = 150
 _DEFAULT_SKYPILOT_IMAGE_GB = 30
 
-_DEFAULT_CPU_IMAGE_ID = 'skypilot:gpu-ubuntu-2204'
-_DEFAULT_GPU_IMAGE_ID = 'skypilot:gpu-ubuntu-2204'
-_DEFAULT_V1_IMAGE_ID = 'skypilot:v1-ubuntu-2004'
+_DEFAULT_CPU_IMAGE_ID = 'skypilot:custom-cpu-ubuntu-v2'
+_DEFAULT_GPU_IMAGE_ID = 'skypilot:custom-gpu-ubuntu-v2'
+_DEFAULT_V1_IMAGE_ID = 'skypilot:custom-gpu-ubuntu-v1'
```
Comment (Michaelvll) [inline sky/clouds/azure.py:44]: Should we add the automatic refetch images catalog logic for azure as well?

Comment (yika-luo) [reply sky/clouds/azure.py:44]: What kind of automatic refetch images? The azure images.csv actually doesn't need to be updated at all

Comment (Michaelvll) [reply sky/clouds/azure.py:44]: Something like this? https://github.com/skypilot-org/skypilot/pull/4127

Comment (yika-luo) [reply sky/clouds/azure.py:44]: Oh you mean local images.csv, got it

**Review by Michaelvll — APPROVED** [review]
LGTM! Thanks @yika-luo!

**Pull request merged**